src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -587,17 +599,10 @@ public EmailWrapper sendRegistrationInviteToStudentAfterGoogleIdReset(String cou
         List<StudentAttributes> studentDataList = getUnregisteredStudentsForCourse(courseId);
         
         List<EmailWrapper> emailsSent = new ArrayList<EmailWrapper>();
-    
-        //TODO: sending mail should be moved to somewhere else.
+        CourseAttributes course = coursesLogic.getCourse(courseId);
         for (StudentAttributes s : studentDataList) {
-            try {
-                EmailWrapper email = sendRegistrationInviteToStudent(courseId, s.email);
-                emailsSent.add(email);
-            } catch (EntityDoesNotExistException e) {
-                Assumption
-                        .fail("Unexpected EntitiyDoesNotExistException thrown when sending registration email"
-                                + TeammatesException.toStringWithStackTrace(e));
-            }
+            EmailWrapper email = scheduleRegistrationInviteToStudent(course, s);
+            emailsSent.add(email);
         }
         return emailsSent;
```

Comment:
It might've been done that way before, but is there a reason why this method needs to return something?